peachjam/forms.py

```diff
@@ -301,3 +314,48 @@ def send_email(self):
             html_message=html,
             fail_silently=False,
         )
+
+
+class FolderForm(forms.ModelForm):
+    class Meta:
+        model = Folder
+        fields = ["name"]
+
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request", None)
+        super().__init__(*args, **kwargs)
+
+        if self.is_bound and self.request:
+            self.data = self.data.copy()
+            if not self.data.get("name") and self.request.htmx:
```

Comment:
It would be a better separation of concerns if the view determined if it was an htmx request and so injected the correct data into the form. The form shouldn't be so closely bound to the request.

Reply:
Or, put another way, the view is responsible for deciding how the user has provided data (either as a POST or part of the htmx request), and then passing that into the form.